lib/from_openstudio/geometry/room.rb

```diff
@@ -70,6 +70,82 @@ def self.properties_from_space(space)
     def self.energy_properties_from_space(space)
       hash = {}
       hash[:type] = 'RoomEnergyPropertiesAbridged'
+      # set room energy properties
+      unless space.defaultConstructionSet.empty?
+        const_set = space.defaultConstructionSet.get
+        hash[:construction_set] = const_set.nameString
+      end
+      unless space.spaceType.empty?
+        space_type = space.spaceType.get
+        hash[:program_type] = space_type.nameString
+      end
+      # TODO: These are loads assigned to the space directly. How should duplicates created in programtype, if any, be handled? 
+      unless space.people.nil?
+        space.people.each do |people|
+          hash[:people] = Honeybee::PeopleAbridged.from_load(people)
+        end
+      end
+      unless space.lights.nil?
+        space.lights.each do |light|
+          hash[:lighting] = Honeybee::LightingAbridged.from_load(light)
+        end
+      end
+      unless space.electricEquipment.nil?
+        space.electricEquipment.each do |elc_eqpmt|
+          hash[:electric_equipment] = Honeybee::ElectricEquipmentAbridged.from_load(elc_eqpmt)
+        end
+      end
+      unless space.gasEquipment.nil?
+        space.gasEquipment.each do |gas_eqpmt|
+          hash[:gas_equipment] = Honeybee::GasEquipmentAbridged.from_load(gas_eqpmt)
+        end
+      end
+      unless space.spaceInfiltrationDesignFlowRates.nil?
+        space.spaceInfiltrationDesignFlowRates.each do |infilt|
+          hash[:infiltration] = Honeybee::InfiltrationAbridged.from_load(infilt)
+        end
+      end
+      unless space.designSpecificationOutdoorAir.empty?
+        space.designSpecificationOutdoorAir.each do |ventilation|
+          hash[:ventilation] = Honeybee::VentilationAbridged.from_load(ventilation)
+        end
+      end
+      unless space.daylightingControls.empty?
+        space.daylightingControls.each do |daylight_control|
+          hash[:daylighting_control] = Honeybee::DaylightingControl.from_load(daylight_control)
+        end
+      end
+      thermal_zone = space.thermalZone
+      unless thermal_zone.empty?
+        thermal_zone = space.thermalZone.get
+        # TODO: The schema for setpoint looks incorrect - there isn't a combined setpoint object in OS. The identifier can't be assigned.
+        # For now using the thermal zone name as setpoint identifier.
+        hash[:setpoint] = {}
+        hash[:setpoint][:identifier] = thermal_zone.nameString
+        unless thermal_zone.thermostatSetpointDualSetpoint.empty?
+          thermostat = thermal_zone.thermostatSetpointDualSetpoint.get
+          #TODO: heating_schedule/cooling_schedule is an optional property in OS Thermostat object. This should not be a required property in the schema.
+          unless thermostat.heatingSetpointTemperatureSchedule.empty?
```

Comment:
@chriswmackey In the OS SDK Heating and Cooling schedule are optional properties for the [thermostat object](https://openstudio-sdk-documentation.s3.amazonaws.com/cpp/OpenStudio-3.3.0-doc/model/html/classopenstudio_1_1model_1_1_thermostat_setpoint_dual_setpoint.html). These are however required properties. 
This will throw a schema validation error for the reverse translators if these aren't set in the OS model.
Any thoughts on how to avoid that?

Reply:
Hm. I see that EnergyPlus also interprets the Thermostat as valid/simulate-able when there are no setpoints on the thermostat object, which seems odd. If a given schedule is missing, can we just create a dummy setpoint schedule with extreme values. For example 100C for the cooling setpoint and -100C for the heating setpoint?